File: 0958-sort-array-by-parity-ii/0958-sort-array-by-parity-ii.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        vector<int> answer(nums.size());
        int evenIndex = 0;   
        int oddIndex = 1;    
        for (int n : nums) {
            if (n % 2 == 0) { 
                answer[evenIndex] = n;
                evenIndex += 2;
            } else { 
                answer[oddIndex] = n;
                oddIndex += 2;
            }
        }

        return answer;
    }
};

File: 0958-sort-array-by-parity-ii/0958-sort-array-by-parity-ii.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        answer=[] 
        even = [n for n in nums if n%2 == 0]
        odd =  [n for n in nums if n%2 == 1]
 
        for a, b in zip(even, odd):
            answer.append(a)
            answer.append(b)
        
        return answer